Tests/SentryTests/Networking/SentryReachabilitySwiftTests.swift

```diff
@@ -10,16 +122,27 @@ final class SentryReachabilitySwiftTests: XCTestCase {
         // observers are adequately synchronized and not if we call
         // SCNetworkReachability correctly.
         sut.skipRegisteringActualCallbacks = true
-        testConcurrentModifications(writeWork: {_ in
-            sut.add(TestReachabilityObserver())
+        testConcurrentModifications(writeWork: { _ in
+            sut.add(TestSentryReachabilityObserver())
         }, readWork: {
             sut.removeAllObservers()
         })
     }
-}
 
-class TestReachabilityObserver: NSObject, SentryReachabilityObserver {
-    func connectivityChanged(_ connected: Bool, typeDescription: String) {
-        // Do nothing
+    /// This tests actually test NWPathMonitor response, if it becomes blaky we can disable it
+    func testRegisteringActualCallbacks_CallbackIsCalled() {
+        reachability.skipRegisteringActualCallbacks = false
+        reachability.setReachabilityIgnoreActualCallback(false)
+        
+        let expectation = XCTestExpectation(description: "Callback should be called")
+        
+        let observer = TestSentryReachabilityObserver()
+        observer.onReachabilityChanged = { _, _ in
+            expectation.fulfill()
+        }
+        
+        reachability.add(observer)
+        
```

Comment:
`m`: I think that doesn't work as we only get an update when the connectivity changes. I think it's OK to remove the XCTestExpectation and only validate that we can add and remove an observer. I guess here it would be OK to put a small delay of a second or so in between those two.